services/src/main/java/org/fao/geonet/api/registries/DirectoryUtils.java

```diff
@@ -430,4 +431,31 @@ private static String search(ServiceContext context, Map<String, String> searchP
         }
         return null;
     }
+
+    public static Element removeUnsedLangsEntry(Element entry, final String[] langs) throws JDOMException {
+        List<Element> multilangElement = (List<Element>)Xml.selectNodes(entry, "*//node()[@locale]");
+
+        List<String> twoCharLangs = new ArrayList<String>();
+        for(String l : langs) {
+            twoCharLangs.add("#" + XslUtil.twoCharLangCode(l).toUpperCase());
+        }
+
+        for(Element el : multilangElement) {
+            if(!twoCharLangs.contains(el.getAttribute("locale").getValue())) {
+                Element parent = (Element)el.getParent();
+                parent.detach();
+            }
+        }
+        return entry;
```

Comment:
If there is only one lang requested, the PT_FreeText element matching this lang should be returned as gco:CharacterString ? So we should send mdOtherLanguages and mdLanguage. We could have one case when no mdLanguage is defined, in this case lang is null and we don't really know how to affect the subtemplate in this case. Removing all PT_FreeText is probably a good option for tihs.